policy-controller/k8s/index/src/index.rs

```diff
@@ -83,6 +87,13 @@ struct Pod {
     /// `Namespace::get_pod_server` when a client discovers a port that has no
     /// configured server (and i.e. uses the default policy).
     port_servers: pod::PortMap<PodPortServer>,
+
+    /// The pod's probe ports and their respective paths.
+    ///
+    /// In order for the policy controller to authorize probes, it must be
+    /// aware of the probe ports and the expected paths on which probes are
+    /// expected.
+    _probes: pod::PortMap<HashSet<String>>,
```

Comment:
I don't think there's really any point to storing this as a `PortMap<HashSet<String>>` since we read this as a list of probes and will eventually use it to construct a list of routes.  I think `Vec<Probe>` where Probe is a struct with a path (`Option<String>`) and a port `NonzeroU16` would be a more natural type.

Reply:
We want the probes to be keyed off of ports here. The reason for this can be explained by how this will be used in `Pod::default_inbound_server`.

In that function, we have a port that we're creating an InboundServer for. When we create the `http_routes`, we need to check the Pod's `_probes` field to see what path's we need to create authorizations for.

Now the Pod could have multiple containers that each expose liveness and readiness probes on multiple ports. We want to check the specific paths for the port that we're looking for (e.g. `_probes.get(port)`).